### PR TITLE
[MOB-71] Migrate event table room

### PR DESCRIPTION
**Pull request by phillipsgoncalves**
What does this PR do?

This PR aims at implementing the event room table.

Database changed?

No

Where should the reviewer start?

 RoomEvent.java
How should this be manually tested?
Try sending events to aptoide bi. Check in charles if they are correctly sent.

What are the relevant tickets?

Tickets related to this pull-request: MOB-71

Questions:

Does this add new dependencies which need to be added? (Eg. new keys, etc.)

Code Review Checklist

 Documentation on public interfaces
 Database changed? If yes - Migration?
 Remove comments & unused code & forgotten testing Logs
 Codestyle
 New Kotlin code has unit tests
 New flows in presenters unit tests
 Mappers/Validators with any kind of logic unit tests
 Functional tests pass

### CODE REVIEW

File: app/src/main/java/cm/aptoide/pt/ApplicationModule.java [hunk 1]
```diff
@@ -1353,15 +1366,15 @@ Retrofit providesLoadTopReactionsRetrofit(@Named("reactions-host") String baseHo
...
+ /* @Singleton @Provides RealmEventMapper providesRealmEventMapper(
       @Named("default") ObjectMapper objectMapper) {
     return new RealmEventMapper(objectMapper);
   }
 
   @Singleton @Provides EventsPersistence providesEventsPersistence(Database database,
       RealmEventMapper mapper) {
     return new RealmEventPersistence(database, mapper);
-  }
+  }*/
```
Comment (jdandrade) [inline app/src/main/java/cm/aptoide/pt/ApplicationModule.java]: comments

File: app/src/main/java/cm/aptoide/pt/ApplicationModule.java [hunk 2]
```diff
@@ -1018,6 +1009,28 @@ AnalyticsBodyInterceptorV7 provideAnalyticsBodyInterceptorV7(
     return new Database();
   }
 
+  @Singleton @Provides AptoideDatabase providesAptoideDataBase() {
+    return Room.databaseBuilder(getApplicationContext(), AptoideDatabase.class,
+        BuildConfig.ROOM_DATABASE_NAME)
+        .fallbackToDestructiveMigration()
```
Comment (jdandrade) [inline app/src/main/java/cm/aptoide/pt/ApplicationModule.java]: This should only be used for testing, we cannot forget to remove before it is merged to dev.

File: aptoide-database/build.gradle [hunk 3]
```diff
@@ -55,4 +63,12 @@ dependencies {
...
   // And ProGuard rules for RxJava!
   implementation "com.artemzin.rxjava:proguard-rules:${RXJAVA_PROGUARD_RULES_VERSION}"
+  implementation 'androidx.room:room-runtime:2.2.3'
+  annotationProcessor 'androidx.room:room-compiler:2.2.3'
+
+  // RxJava support for Room
+  implementation "androidx.room:room-rxjava2:${ROOM_VERSION}"
+
+  implementation "io.reactivex.rxjava2:rxjava:2.2.10"
+  implementation "io.reactivex.rxjava2:rxandroid:2.1.1"
```
Comment (jdandrade) [inline aptoide-database/build.gradle]: Any reason not to use project level variables here?

Comment (phillipsgoncalves) [reply aptoide-database/build.gradle]: already changed it in another branch

**Pull request merged**